src/openrct2/rct12/SawyerChunkReader.h

```diff
@@ -36,15 +36,17 @@ namespace OpenRCT2
 
 /**
  * Reads sawyer encoding chunks from a data stream. This can be used to read
- * SC6, SV6 and RCT2 objects.
+ * SC6, SV6 and RCT2 objects. persistentChunks is a hint to the reader that the chunk will be preserved,
+ * and thus the chunk memory should be shrunk.
  */
 class SawyerChunkReader final
 {
 private:
     OpenRCT2::IStream* const _stream = nullptr;
+    const bool _createsPersistentChunks = false;
 
 public:
-    explicit SawyerChunkReader(OpenRCT2::IStream* stream);
+    explicit SawyerChunkReader(OpenRCT2::IStream* stream, bool persistentChunks = false);
```

Comment:
This argument is never set to true. This essentially remove sthe persistent chunks entirely. Doesn't this make loading of parks slower every time now?

Reply:
No, the opposite - unless I overlooked something, the codebase has no instances where a loaded `SawyerChunk` is persisted, they are always kept as short lived local variables. Therefore, with `persistentChunks` set to `false`, the code won't bother shrinking allocations as they are freed microseconds later anyway.

Reply:
If this is not used then I'm for removing it

Reply:
Both are fine for me - if you think it's better this way, I can remove those reallocate functions completely.